src/main/java/com/mcmoddev/mmdbot/modules/logging/misc/ScamDetector.java

```diff
@@ -0,0 +1,156 @@
+            msg.delete().reason("Scam link").queue($ -> {
+                executeInLoggingChannel(channel -> channel.sendMessageEmbeds(embed).queue());
+                mute(guild, member);
+            });
+        }
+    }
+
+    @Override
+    public void onGuildMessageUpdate(@NotNull final GuildMessageUpdateEvent event) {
+        final var msg = event.getMessage();
+        final var member = msg.getMember();
+        if (member == null || msg.getAuthor().isBot() || msg.getAuthor().isSystem() ||
+            member.hasPermission(Permission.MANAGE_CHANNEL)) {
+            return;
+        }
+        if (containsScam(msg)) {
+            final var guild = msg.getGuild();
+            final var embed = getLoggingEmbed(msg, ", by editing an old message");
+            msg.delete().reason("Scam link").queue($ -> {
+                executeInLoggingChannel(channel -> channel.sendMessageEmbeds(embed).queue());
+                mute(guild, member);
+            });
+        }
+    }
+
+    private static void mute(final Guild guild, final Member member) {
+        final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+        // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+        final var mutedRole = guild.getRoleById(mutedRoleID);
+        if (mutedRole == null) {
+            MMDBot.LOGGER.error(MMDMarkers.MUTING, "Unable to find muted role {}", mutedRoleID);
+            return;
+        }
+        guild.addRoleToMember(member, mutedRole).queue();
+    }
+
+    private static MessageEmbed getLoggingEmbed(final Message message, final String extraDescription) {
+        final var member = message.getMember();
+        return new EmbedBuilder().setTitle("Scam link detected!")
+            .setDescription(String.format("User %s sent a scam link in %s%s. Their message was deleted, and they were muted.", member.getUser().getAsTag(),
+                message.getTextChannel().getAsMention(), extraDescription))
+            .addField("Message Content", MarkdownUtil.codeblock(message.getContentRaw()), false)
+            .setColor(Color.RED)
+            .setTimestamp(Instant.now())
+            .setFooter("User ID: " + member.getIdLong())
+            .setThumbnail(member.getEffectiveAvatarUrl()).build();
+    }
+
+    private static void executeInLoggingChannel(Consumer<TextChannel> channel) {
+        Utils.getChannelIfPresent(MMDBot.getConfig().getChannel("events.requests_deletion"), channel);
+    }
+
+    public static boolean containsScam(final Message message) {
+        final String msgContent = message.getContentRaw().toLowerCase(Locale.ROOT);
+        synchronized (SCAM_LINKS) {
+            for (final var link : SCAM_LINKS) {
+                if (msgContent.contains(link)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static void setupScamLinks() {
+        SCAM_LINKS.clear();
```

Comment:
Ideally, the call to `clear` should be done just before `addAll`, to prevent a situation where the scam links set is empty afterwards a failed refresh due to e.g. connection or API issues.